### PR TITLE
add(layout): Box component

### PR DESCRIPTION
**Pull request by kirill-kruchkov**
## Purpose

Box component has no UI and specifies paddings and spacing between child components.

### Visualization

image

### Checklist

- [x] Update changelog

### CODE REVIEW

File: src/layout/Stack/Stack.js [hunk 1]
```diff
@@ -44,5 +44,8 @@ Stack.propTypes = {
   direction: PropTypes.oneOf(['horizontal', 'vertical']),
 };
 
-/** @component */
+/**
+ * @component
+ * @deprecated please use `Box` component.
```
Comment (adrianosmond) [inline src/layout/Stack/Stack.js]: This deprecation should probably also be documented in Stack.md

Comment (adrianosmond) [inline src/layout/Stack/Stack.js]: This deprecation should probably also be reflected in Stack.md

File: src/layout/Box/Box.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  direction,
+  children,
+  ...otherProps
+}) => React.createElement(renderAs, otherProps, children);
+
+Box.defaultProps = {
+  renderAs: 'div',
+  direction: 'vertical',
+  padding: '0',
+  stack: '0',
```
Comment (adrianosmond) [inline src/layout/Box/Box.js]: It wasn't immediately obvious to me that the `stack` props were controlling the spacing of the items in the stack. That might just be me... How would you feel about calling them `spacing` or `innerSpacing` or `stackSpacing` to make it more explicit that that's what you're controlling here?

**Comment by kirill-kruchkov** [comment]
@dinkra I'll fix it!

File: src/layout/Box/Box.js (hunk 2 above)
Comment (kirill-kruchkov) [reply src/layout/Box/Box.js]: @dinkra @frontalicious-martijn @zoheiry WDYT?

Comment (dinkra) [reply src/layout/Box/Box.js]: I think if we have `stack` prop description it should be fine.

File: src/layout/Stack/Stack.js (hunk 1 above)
Comment (kirill-kruchkov) [reply src/layout/Stack/Stack.js]: Added to MD

File: src/layout/Box/Box.js (hunk 2 above)
Comment (kirill-kruchkov) [reply src/layout/Box/Box.js]: Fixed the props thing:

image

**Pull request merged**